lib/request-helpers.js

```diff
@@ -0,0 +1,27 @@
+/**
+ *
+ * @param {import('./typings.js').RequestWithSubnet} request
+ * @param {import('fastify').FastifyReply} reply
+ */
+export const subnetPreHandlerHook = async (request, reply) => {
+  // @ts-ignore TODO: Fix
+  const { subnet } = request.params
+  if (!subnet || typeof subnet !== 'string') {
+    reply.code(400).send({ error: 'Invalid subnet parameter' })
+    return
+  }
+
+  if (!isValidSubnet(subnet.toLowerCase())) {
+    reply.code(404).send()
```

Comment:
In my experience with cid.contact that returns 404 for unknown CIDs, it's difficult to tell whether you have a typo in the requested path or whether you used invalid value for one of the path parameters.

For example, both `POST /walrus/measurment` and `POST /filecoin/measurement` return the same code 404. The first one contains a typo in the last path segment, the second is using an unsupported subnet name.

Let's return the status code 400 instead. You already return 400 when the subnet is not provided.

```suggestion
    reply.code(400).send()
```